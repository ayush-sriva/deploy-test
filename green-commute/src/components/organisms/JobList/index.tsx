import { Grid, styled } from '@mui/material';
import responsiveTheme from '../../../theme';
import TypographyComponent from '../../atoms/Typography';
import ButtonComponent from '../../atoms/Button';
import ChipItem from '../../molecules/ChipItem';

interface JobListProps {
  filters?: string[];
  handleDelete?: React.MouseEventHandler<HTMLDivElement>;
  handleClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const Heading = styled(TypographyComponent)({
  color: responsiveTheme.palette.secondary.dark,
  paddingBottom: '0.25rem',
});

const SubHeading = styled(TypographyComponent)({
  color: responsiveTheme.palette.secondary.dark,
  paddingBottom: '0.75rem',
});

const FiltersContainer = styled(Grid)({
  maxWidth: '34rem',
});

const ButtonContainer = styled(Grid)({
  marginTop: '0.22rem',
  textDecoration: 'underline',
  textDecorationColor: `${responsiveTheme.palette.primary.dark}`,
  textDecorationThickness: '2px',
  textUnderlineOffset: '2px',
});

const renderFilters = (filters: string[], handleDelete: React.MouseEventHandler<HTMLDivElement>) => {
  console.log(filters);
  if (filters !== [] && handleDelete !== null) {
    return filters.map((filter: string) => {
      return (
        <Grid item>
          <ChipItem label={filter} handleDelete={handleDelete} color={'secondary'} />
        </Grid>
      );
    });
  }
};

const JobList: React.FC<JobListProps> = ({ filters, handleDelete, handleClick }) => {
  return (
    <Grid container>
      <Grid item xs={12}>
        <Heading variant='h2'>Job list</Heading>
      </Grid>
      <Grid item xs={12}>
        <SubHeading variant='body2'>Based on your search</SubHeading>
      </Grid>
      <FiltersContainer container>
        <Grid item xs={11}>
          <Grid container rowSpacing={2}>
            {filters && handleDelete && renderFilters(filters, handleDelete)}
          </Grid>
        </Grid>
        <ButtonContainer item xs={1}>
          {filters && filters.length !== 0 && (
            <ButtonComponent
              disableRipple={true}
              buttonVariant='text'
              size='medium'
              type='reset'
              typoVariant='body1'
              children='Clear All'
              color={'error'}
              onClickHandler={handleClick}
              style={{
                width: '80px',
              }}
            />
          )}
        </ButtonContainer>
      </FiltersContainer>
    </Grid>
  );
};

export default JobList;
