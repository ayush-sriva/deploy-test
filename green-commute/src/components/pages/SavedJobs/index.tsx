/**
 * @author Bhavani Somanchi <bhavani.somanchi@zemosolabs.com>
 */

import TemplateHomePage from '../../templates/HomePage';
import Header from '../../organisms/Header';
import SideMenu from '../../organisms/SideMenu';
import { Grid, styled } from '@mui/material';
import Typography from '../../atoms/Typography';
import theme from '../../../theme';
import { useState, useEffect } from 'react';
import JobCard from '../../organisms/FindCard';
import { fetchAllSavedJobs } from '../../../service/JobService';
import { userLocation } from '../../../utils/constants';

const MainContainer2 = styled(Grid)({
  marginTop: '30px',
  [theme.breakpoints.up('lg')]: {
    marginTop: '6rem',
    paddingLeft: '19rem',
  },
  [theme.breakpoints.up('xl')]: {
    marginTop: '30px',
    paddingLeft: '0',
  },
});
const JobsContainer2 = styled(Grid)({
  minHeight: '100vh',
  [theme.breakpoints.up('lg')]: {
    paddingLeft: '19rem',
  },
  [theme.breakpoints.up('xl')]: {
    paddingLeft: '0',
    marginBottom: '5rem',
  },
});
const JobGrid2 = styled(Grid)({
  marginRight: '15px',
  [theme.breakpoints.up('sm')]: {
    marginTop: '1rem',
    width: '576px !important',
    height: '159px ',
    borderRadius: '12px',
  },
  [theme.breakpoints.up('xl')]: {
    marginTop: '1rem',
  },
});

const NewJobCard2 = styled(JobCard)({
  [theme.breakpoints.up('xl')]: {
    padding: '0.5rem',
  },
});

interface CompanyDetails {
  company: string;
  title: string;
  location: string;
  companyLogo: string;
  postedDate: string;
}
interface Job {
  id: string;
  jobTitle: string;
  company: string;
  datePosted: string;
  location: string;
  logo: string;
  description: string;
  aboutCompany: string;
  details: CompanyDetails;
}

const SavedJobsPage = () => {
  const [jobData2, setJobData2] = useState<Array<Job>>([]);

  const getJobData2 = async () => {
    fetchAllSavedJobs().then((jobsData) => {
      setJobData2(jobsData);
    });
  };

  useEffect(() => {
    getJobData2();
  }, []);

  const renderDescription2 = () => {
    return jobData2.map((jobData) => {
      return (
        <JobGrid2 key={jobData.id} item>
          <NewJobCard2
            id={jobData.id}
            company={jobData.company}
            title={jobData.jobTitle}
            location={jobData.location}
            square={false}
            companyLogo={jobData.logo}
            postedDate={jobData.datePosted}
          />
        </JobGrid2>
      );
    });
  };

  return (
    <TemplateHomePage
      header={<Header userLocation={userLocation} />}
      sideMenu={<SideMenu toggler={()=>{}}/>}
      data-testid='home-template'
    >
      <MainContainer2 container direction='row'>
        <Grid item xs={9} lg={9}>
          <Typography variant='h2'>Saved Jobs</Typography>
        </Grid>
      </MainContainer2>
      <JobsContainer2 container>
        <Grid container direction='row'>
          <Grid item xs={12}>
            <Grid container>{renderDescription2()}</Grid>
          </Grid>
        </Grid>
      </JobsContainer2>
    </TemplateHomePage>
  );
};

export default SavedJobsPage;
