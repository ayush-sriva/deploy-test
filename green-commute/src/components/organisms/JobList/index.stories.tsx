import { ComponentStory, ComponentMeta } from '@storybook/react';
import JobList from './index';
import { ReactComponent as savedJobs } from '../../../../public/assets/sideMenuIcons/SavedJobs.svg';

export default {
  title: 'Components/Organisms/JobList',
  component: JobList,
} as ComponentMeta<typeof JobList>;

const filters = ['10 - 20kms', 'Past 24 hours', 'Full time', 'Freshers', 'Green Commute Routes'];

const handleDelete = () => {
  console.log('You have clicked delete');
};

const handleClick = () => {
  filters.splice(0, filters.length);
  console.log('This button is clicked');
};

const Template: ComponentStory<typeof JobList> = (args) => (
  <div style={{ backgroundColor: '#E5E5E5', width: '571px', padding: '5px' }}>
    <JobList {...args} />
  </div>
);

export const Primary = Template.bind({});

Primary.args = {
  filters: filters,
  handleDelete: handleDelete,
  handleClick: handleClick,
};
