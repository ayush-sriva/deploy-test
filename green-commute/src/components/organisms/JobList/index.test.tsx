
import React, { useState } from 'react';
import {render, screen} from '@testing-library/react';
import { ReactComponent as savedJobs } from '../../../../public/assets/sideMenuIcons/SavedJobs.svg';
import renderer from 'react-test-renderer';

import JobList from '.';

const filters = ["10 - 20kms", "Past 24 hours", "Full time", "Freshers", "Green Commute Routes"]

const handleDelete = () => {
    console.log("You have clicked delete")
  };

describe('Organisms', () => {
    test('renders Job list', () => {
        render(<JobList filters={filters} handleDelete={handleDelete} />);
        let button = screen.getAllByRole('button');
        expect(button).toHaveLength(6);
    })
});


it('renders correctly when filter has no items', () => {
    const tree = renderer.create(<JobList />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when filter has items', () => {
    const tree = renderer.create(<JobList filters={filters}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


