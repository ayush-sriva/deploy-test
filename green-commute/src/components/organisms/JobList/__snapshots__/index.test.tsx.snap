// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when filter has items 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-6jajho-MuiTypography-root"
    >
      Job list
    </h2>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 css-t87ivq-MuiTypography-root"
    >
      Based on your search
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container css-lx13dv-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1jzi3sl-MuiGrid-root"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-16gjb1v-MuiGrid-root"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-10fi992-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="reset"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Clear All
        </p>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly when filter has no items 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-6jajho-MuiTypography-root"
    >
      Job list
    </h2>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 css-t87ivq-MuiTypography-root"
    >
      Based on your search
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container css-lx13dv-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1jzi3sl-MuiGrid-root"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-16gjb1v-MuiGrid-root"
    />
  </div>
</div>
`;
